PublicAPI/Microsoft.ApplicationInsights.dll/net45/PublicAPI.Unshipped.txt

```diff
@@ -33,5 +33,237 @@ Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing.IHeartbeatPro
+const Microsoft.ApplicationInsights.MetricDimensionNames.TelemetryContext.Session.IsFirst = "TelemetryContext.Session.IsFirst" -> string
+const Microsoft.ApplicationInsights.MetricDimensionNames.TelemetryContext.User.AccountId = "TelemetryContext.User.AccountId" -> string
+const Microsoft.ApplicationInsights.MetricDimensionNames.TelemetryContext.User.AuthenticatedUserId = "TelemetryContext.User.AuthenticatedUserId" -> string
+const Microsoft.ApplicationInsights.MetricDimensionNames.TelemetryContext.User.Id = "TelemetryContext.User.Id" -> string
+const Microsoft.ApplicationInsights.MetricDimensionNames.TelemetryContext.User.UserAgent = "TelemetryContext.User.UserAgent" -> string
+const Microsoft.ApplicationInsights.Metrics.MetricIdentifier.MaxDimensionsCount = 10 -> int
+const Microsoft.ApplicationInsights.Metrics.MetricSeriesConfigurationForMeasurement.AggregateKindConstants.Constants.AggregateKindDataKeys.Count = "Count" -> string
+const Microsoft.ApplicationInsights.Metrics.MetricSeriesConfigurationForMeasurement.AggregateKindConstants.Constants.AggregateKindDataKeys.Max = "Max" -> string
+const Microsoft.ApplicationInsights.Metrics.MetricSeriesConfigurationForMeasurement.AggregateKindConstants.Constants.AggregateKindDataKeys.Min = "Min" -> string
+const Microsoft.ApplicationInsights.Metrics.MetricSeriesConfigurationForMeasurement.AggregateKindConstants.Constants.AggregateKindDataKeys.StdDev = "StdDev" -> string
+const Microsoft.ApplicationInsights.Metrics.MetricSeriesConfigurationForMeasurement.AggregateKindConstants.Constants.AggregateKindDataKeys.Sum = "Sum" -> string
+const Microsoft.ApplicationInsights.Metrics.MetricSeriesConfigurationForMeasurement.AggregateKindConstants.Constants.AggregateKindMoniker = "Microsoft.Azure.Measurement" -> string
+const Microsoft.ApplicationInsights.Metrics.MetricSeriesConfigurationForMeasurement.Constants.AggregateKindDataKeys.Count = "Count" -> string
+const Microsoft.ApplicationInsights.Metrics.MetricSeriesConfigurationForMeasurement.Constants.AggregateKindDataKeys.Max = "Max" -> string
+const Microsoft.ApplicationInsights.Metrics.MetricSeriesConfigurationForMeasurement.Constants.AggregateKindDataKeys.Min = "Min" -> string
+const Microsoft.ApplicationInsights.Metrics.MetricSeriesConfigurationForMeasurement.Constants.AggregateKindDataKeys.StdDev = "StdDev" -> string
+const Microsoft.ApplicationInsights.Metrics.MetricSeriesConfigurationForMeasurement.Constants.AggregateKindDataKeys.Sum = "Sum" -> string
+const Microsoft.ApplicationInsights.Metrics.MetricSeriesConfigurationForMeasurement.Constants.AggregateKindMoniker = "Microsoft.Azure.Measurement" -> string
+override Microsoft.ApplicationInsights.Metrics.MetricConfiguration.Equals(object obj) -> bool
+override Microsoft.ApplicationInsights.Metrics.MetricConfiguration.GetHashCode() -> int
+override Microsoft.ApplicationInsights.Metrics.MetricIdentifier.Equals(object otherObj) -> bool
```

Comment:
Judging by name `MetricIdentifier` sounds like too low level of a detail

Reply:
Could you please help me understand your question? Do you have concerns with the name "MetricIdentifier"? Do you have concerns with the usage of the type `MetricIdentifier`? Something else? The level of detail is to low for what specifically? What detail exactly do you refer to?

In order to provide as much context as possible, let me please describe the purpose and the usage of the `MetricIdentifier` class.

A metric is a grouping of one or more metric data series. Besides the actual data (aka the metric series) such a group has 2 attributes:

i) The *metric configuration*. It is a set of settings that define the behavior of the group and the contained data. For example, it specifies how values are aggregated, how many series are allowed in the group and so on. Metrics with the same configuration may contain different data, but they will exhibit the same  behavior. You can think of this as loosely corresponding to a "class type" for a metric.

ii) The *metric identifier*. It is a set of settings that define the identity of the group so that it can be differentiated from any other groups (aka from other metrics). The identifier includes the *metric name*, the *metric namespace*, as well as the number and the names of the *metric dimensions*. An identifier uniquely identifies a metric within the respective scope. You can think of this as loosely corresponding to a "class instance id" of a metric.

Please let me know if this answers your question.
If you require adjustments to these APIs, please request *precisely* the changes you would like to see so that I can make them.
Thank you.